From: John Johansen <john.johansen@canonical.com>
Date: Thu, 19 Feb 2026 09:36:18 -0800
Subject: [PATCH 3/3] parser: restrict umask to allow only user permissions

The umask is an inverted permissions mask. The bits that are set
are what is not allowed. So invert ~S_IRWXU.

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 parser/policy_cache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser/policy_cache.c b/parser/policy_cache.c
index cc6a9f5..232431e 100644
--- a/parser/policy_cache.c
+++ b/parser/policy_cache.c
@@ -237,7 +237,7 @@ int setup_cache_tmp(const char **cachetmpname, char *cachename, int create_compr
 			perror("asprintf");
 			return -1;
 		}
-		prev_mask = umask(S_IRWXU);
+		prev_mask = umask(~S_IRWXU);
 		if ((cache_fd = mkstemp(tmpname)) < 0) {
 			perror("mkstemp");
 			umask(prev_mask);
